packages/@aws-cdk/example-construct-library/lib/example-resource.ts

```diff
@@ -0,0 +1,501 @@
+/*
+ * We always import other construct libraries entirely with a prefix -
+ * we never import individual classes from them without a qualifier
+ * (the prefix makes it more obvious where a given dependency comes from,
+ * and prevents conflicting names causing issues).
+ * Our linter also enforces ES6-style imports -
+ * we don't use TypeScript's import a = require('a') imports.
+ */
+import * as cloudformation from '@aws-cdk/aws-cloudformation';
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as events from '@aws-cdk/aws-events';
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as core from '@aws-cdk/core';
+// for files that are part of this package, we do import individual classes or functions
+import { exampleResourceArnComponents } from './example-resource-utils';
+
+/**
+ * The interface that represents the ExampleResource resource.
```

Comment:
Fucking AMAZING job on all the explanations... but I wonder if this is the best place for them.

My experience with explanatory-comments-in-templates is that after a while people just start skipping them and leaving them in (instead of deleting them). So my gut says this all shouldn't live here... unfortunately I don't have a better solution.

Best I can come up with is put all the explanation (basically the Construct Library Guide) up on the GitHub wiki, and putting links to the relevant sections into the comment blocks...

?